src/problem_bank_scripts/problem_bank_scripts.py

```diff
@@ -710,7 +710,10 @@ def process_matching(part_name, parsed_question, data_dict):
         if "option" in a:
             value = f"|@|@ params.{part_name}.{a}.value @|@|"
 
-            options += f"\t<pl-option> {value} </pl-option>\n"
+            if name := data_dict["params"][f"{part_name}"][a].get("name"):
+                options += f"\t<pl-option name='{name}' > {value} </pl-option>\n"
```

Comment:
Should be something like:
name = f"|@|@ params.{part_name}.{a}.name @|@|"
options += f"\t<pl-option name='{name}' > {value} </pl-option>\n"

Reply:
@Bluesy1 Now that I think about it, hard coding the name value in might cause problems in terms of randomization when different variants have different name values. There isn't a problem in my case because both variants have the exact same option names.

Reply:
I see know what you're saying - it should be like this instead of what I did?
```suggestion
                name = f"|@ params.{part_name}.{a}.name @|"
                options += f"\t<pl-option name='{name}' > {value} </pl-option>\n"
```
(I'm only using `|@...@|` instead of `|@|@...@|@|` since we don't need to support arbitrary html here)
Yeah this is reasonable, not sure how I didn't think about that when I wrote this change. I'm not sure when people would want truely randomized option names, since they don't appear to the student, and they are only needed if the option is a correct match (and if you have randomized names instead of static ones, I don't see how its really all that easier to use name over just setting match to the randomized option, but I'll chalk that up to there being multiple approaches to problems and I can't see every version), but it should be dynamic.

Reply:
Yes that is what I meant. I think when you have randomization, you have to make sure that the number of options and statements are the same across all your variants in the same question, which I personally find easier to do by defining each option explicitly, and referring to them using a key.

Reply:
Makes sense - I'd be fine with that change, as it still keeps the optionality and its what should have been done in this PR in hindsight